docker/docker.go

```diff
@@ -264,6 +265,30 @@ func WithSharedDirectory(hostPath, containerPath string) ConfigOption {
 }
 
 func withVolume(typ mount.Type, hostPath, containerPath string) ConfigOption {
+	c, err := client.NewEnvClient()
+	if err != nil {
+		logrus.Fatalf("could not create docker client")
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+	defer cancel()
+
+	info, err := c.Info(ctx)
+	if err != nil {
+		logrus.Fatalf("could not get information")
```

Comment:
fatal feels a bit too drastic, this will kill the daemon. If we can I think it would be nice to return a nice error with the reason why the container could not be created.

Reply:
you're right, it seems a bit drastic. anyway, I then need to change the return value to `(ConfigOption, error)`. But what about the other functions returning a `ConfigOption`? Maybe for consistency, we should change them too even if they don't raise any potential error.

Reply:
:thinking: yes for consistency, but then everything gets flooded with `if err != nil`... I think for now we can modify the return values only for those that can return errors.

Or, if returning an error here makes the code uglier, you could also make the path translation as a separate step. Something like this:
```go
	indexDir := filepath.Join(s.datadir, "gitbase", s.workdirHash)

	workdirHostPath, err := docker.HostPath(s.workdir)
	if err != nil {
	}

	indexHostPath, err := docker.HostPath(indexDir)
	if err != nil {
	}

	return Component{
		Name: gitbase.Name,
		Start: createGitbase(
			docker.WithSharedDirectory(workdirHostPath, gitbaseMountPath),
			docker.WithSharedDirectory(indexHostPath, gitbaseIndexMountPath),
			docker.WithPort(gitbasePort, gitbasePort),
		),
		Dependencies: []Component{
			s.bblfshComponent(),
		},
	}
```

Reply:
Addressed [here](https://github.com/src-d/engine/pull/235/commits/0db388d158478a338ef9c371f79e278d6633876c). The code has now more `if err != nil`, but I think is better as-is as it surely provides a more informative error.